hack/docker.sh

```diff
@@ -5,6 +5,21 @@ set -o nounset
 set -o pipefail
 
 # This script holds docker related functions.
+# You can set the platform to build with BUILD_PLATFORMS, with format: `<os>/<arch>`
+#
+# Usage:
+#   hack/docker.sh <target>
+# Args:
+#   $1:              target to build
+#   BUILD_PLATFORMS: platforms to build. You can set one or more platforms separated by comma.
+#                    e.g.: linux/amd64,linux/amd64
+#   OUTPUT_TYPE      Destination to save image(docker/registry).
+#   REGISTRY         image registry
+#   VERSION          image version
+# Examples:
+#   hack/docker.sh karmada-aggregated-apiserver
+#   BUILD_PLATFORMS=linux/amd64,linux/amd64 hack/docker.sh karmada-aggregated-apiserver
```

Comment:
```
# BUILD_PLATFORMS=linux/amd64,linux/amd64 hack/docker.sh karmada-aggregated-apiserver
Cross building image for linux/amd64,linux/amd64: swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-aggregated-apiserver:unknown
+ docker buildx build --output=type=docker --platform linux/amd64,linux/amd64 --build-arg BINARY=karmada-aggregated-apiserver --tag swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-aggregated-apiserver:unknown --file hack/../cluster/images/buildx.Dockerfile hack/../_output/bin
[+] Building 0.0s (0/0)                                                                                                                                                      
error: docker exporter does not currently support exporting manifest lists
```

Reply:
em, it's my fault. We cannot buildx multi platform with `--output=type=docker`. As this script is not faced to user(user shall use `make image` or `make mp`), so i didn't add validate here. I will change this doc.